breacharbiter.go

```diff
@@ -211,32 +249,37 @@ func (b *breachArbiter) contractObserver(activeChannels []*lnwallet.LightningCha
 out:
 	for {
 		select {
-		case breachInfo := <-b.breachedContracts:
+		case pendingRetribution := <-b.breachedContracts:
 			// A new channel contract has just been breached! We
 			// first register for a notification to be dispatched
 			// once the breach transaction (the revoked commitment
 			// transaction) has been confirmed in the chain to
 			// ensure we're not dealing with a moving target.
-			breachTXID := &breachInfo.commitHash
+			breachTXID := &pendingRetribution.commitHash
 			confChan, err := b.notifier.RegisterConfirmationsNtfn(breachTXID, 1)
 			if err != nil {
-				brarLog.Errorf("unable to register for conf for txid: %v",
-					breachTXID)
+				brarLog.Errorf("unable to register for conf updates for txid: "+
+					"%v, err: %v", breachTXID, err)
 				continue
 			}
 
-			brarLog.Warnf("A channel has been breached with tx: %v. "+
+			brarLog.Warnf("A channel has been breached with txid: %v. "+
 				"Waiting for confirmation, then justice will be served!",
 				breachTXID)
 
-			// With the notification registered, we launch a new
-			// goroutine which will finalize the channel
-			// retribution after the breach transaction has been
-			// confirmed.
+			// Persist the pending retribution state to disk.
```

Comment:
I'd say that persisting the retribution state to disk should be the very _first_ action taken within the body of this select case.